jwst/rscd/rscd_sub.py

```diff
@@ -199,91 +321,101 @@ def get_rscd_parameters(input_model, rscd_model):
-                                      rowtype == 'EVEN'))))
-
-    # Check for no match found for EVEN
-    if len(index[0]) == 0:
-        log.warning('No matching row found in RSCD reference table for')
-        log.warning('READPATT=%s, SUBARRAY=%s, Row type=EVEN',
-                    readpatt, subarray)
-        return None
-
-    # Load the params from the matching table row
-    param = {}
-    param['even'] = {}
-    param['odd'] = {}
-
-    param['even']['tau'] = tau_table[index]
-    param['even']['ascale'] = ascale_table[index]
-    param['even']['pow'] = pow_table[index]
-    param['even']['illum_zp'] = illum_zp_table[index]
-    param['even']['illum_slope'] = illum_slope_table[index]
-    param['even']['illum2'] = illum2_table[index]
-    param['even']['param3'] = param3_table[index]
-    param['even']['crossopt'] = crossopt_table[index]
-    param['even']['sat_zp'] = sat_zp_table[index]
-    param['even']['sat_slope'] = sat_slope_table[index]
-    param['even']['sat2'] = sat_2_table[index]
-    param['even']['sat_mzp'] = sat_mzp_table[index]
-    param['even']['sat_rowterm'] = sat_rowterm_table[index]
-    param['even']['sat_scale'] = sat_scale_table[index]
-
-    # Find the matching table row index for ODD row
-    index2 = np.asarray(np.where(np.logical_and(readpatt_table == readpatt,
-                        np.logical_and(subarray_table == subarray,
-                                       rowtype == 'ODD'))))
-
-    # Check for no match found ODD row
-    if len(index[0]) == 0:
-        log.warning('No matching row found in RSCD reference table for')
-        log.warning('READPATT=%s, SUBARRAY=%s, Row type=ODD',
-                    readpatt, subarray)
-        return None
-
-    # Load the params from the matching table row
-    param['odd']['tau'] = tau_table[index2]
-    param['odd']['ascale'] = ascale_table[index2]
-    param['odd']['pow'] = pow_table[index2]
-    param['odd']['illum_zp'] = illum_zp_table[index2]
-    param['odd']['illum_slope'] = illum_slope_table[index2]
-    param['odd']['illum2'] = illum2_table[index2]
-    param['odd']['param3'] = param3_table[index2]
-    param['odd']['crossopt'] = crossopt_table[index2]
-    param['odd']['sat_zp'] = sat_zp_table[index2]
-    param['odd']['sat_slope'] = sat_slope_table[index2]
-    param['odd']['sat2'] = sat_2_table[index2]
-    param['odd']['sat_mzp'] = sat_mzp_table[index2]
-    param['odd']['sat_rowterm'] = sat_rowterm_table[index2]
-    param['odd']['sat_scale'] = sat_scale_table[index2]
+    # read table 1: containing the number of groups to skip
+    for tabdata in rscd_model.rscd_group_skip_table:
+        subarray_table = tabdata['subarray']
+        readpatt_table = tabdata['readpatt']
+        group_skip_table = tabdata['group_skip']
+        group_skip_table = int(group_skip_table)
+        if (subarray_table == subarray and readpatt_table == readpatt):
+            param['skip'] = group_skip_table
```

Comment:
Do you want a `break` here?  Or can more than one row match?

Reply:
Yes.  I've also added breaks for the other for loops in this function.